### ISSUE TITLE
Question for the regularization_type

### ISSUE THREAD

**Issue by SYSUykLin** [Anttwo/SuGaR#126]
If I choose the density regularization，what regularization will be used in this code? I did not find this part on your paper. Thanks

**Comment by Anttwo** [comment]
Hello @SYSUykLin,

This is an interesting question!
It's true that the density regularization is quite hidden in the paper.
Please take a look at page 5, just under Equation 5, for more details.

The density regularization consists in a direct comparison between the real density function $d$ and the "ideal" density function $\bar{d}$:

$$
\cal{R}_{\text{density}} =  \frac{1}{\cal{P}}  \sum_p | d(p) - \bar{d}(p) |
$$

The $\cal{R}_{\text{Norm}}$ term is actually the same than for SDF, as the gradient of density and SDF are colinear anyway.

**Comment by SYSUykLin** [comment]
Thanks for your reply. I see. Another question, How many GS points are there during the Refine stage? 6 points per triangle, that is a huge number.

**Comment by Anttwo** [comment]
Vanilla Gaussian Splatting generally ends up with 1M to 5M Gaussians after optimization.
Similarly, during the refinement stage of SuGaR, we aim for approximately 2M Gaussians on the surface, as it is generally enough to get a good looking representation of real scenes.

Therefore, we increase the number of surface-aligned Gaussians per triangle when we decrease the number of triangles.

For example, when using 1M vertices (this corresponds to the `--high_poly` config), the mesh generally has approximately 2M triangles, so we use only 1 Gaussian per triangle.
When using only 200k vertices (this corresponds to the `--low_poly` config), the mesh generally has approximately 400k triangles, so we use 6 Gaussians per triangle.

In the end, SuGaR always has a similar number of Gaussians than vanilla 3DGS.

**Comment by SYSUykLin** [comment]
Thanks for your reply. Greet paper. And last question, I try to normalize the vertex at the coarse mesh.
It means in sugar class of the sugar_model.py line 208,  I add the normalized code for mesh vertex:
`points = torch.tensor(np.array(surface_mesh_to_bind.vertices)).float().to(nerfmodel.device)`
I add the code:
`vmax, vmin = points.max(dim=0), points.min(dim=0)
        scale = 2 / torch.max(vmax - vmin).item()
        points = points - (vmax + vmin) / 2 # Center mesh on origin
        points = points * scale                  # Rescale to unit size

`
And the loss is unstability. May I ask why? This confuses me all night, I think this mimic modification will not change the result. Thanks
